cli/command/load_balancer.go

```diff
@@ -10,6 +18,497 @@ func NewLoadBalancerCommand(b *CommandBase) *LoadBalancerCommand {
+	}
+
+	loadBalancerID, err := l.resolveSingleEntityIDHelper("load_balancer", args["LOADBALANCER_NAME"])
+	if err != nil {
+		return err
+	}
+
+	loadBalancer, err := l.client.ReadLoadBalancer(loadBalancerID)
+	if err != nil {
+		return err
+	}
+
+	if requiresUpdate {
+		if healthCheck.Target != "" {
+			loadBalancer.HealthCheck.Target = healthCheck.Target
+		}
+
+		if healthCheck.Interval != 0 {
+			loadBalancer.HealthCheck.Interval = healthCheck.Interval
+		}
+
+		if healthCheck.Timeout != 0 {
+			loadBalancer.HealthCheck.Timeout = healthCheck.Timeout
+		}
+
+		if healthCheck.HealthyThreshold != 0 {
+			loadBalancer.HealthCheck.HealthyThreshold = healthCheck.HealthyThreshold
+		}
+
+		if healthCheck.UnhealthyThreshold != 0 {
+			loadBalancer.HealthCheck.UnhealthyThreshold = healthCheck.UnhealthyThreshold
+		}
+
+		req := models.UpdateLoadBalancerRequest{
+			LoadBalancerID: loadBalancerID,
+			Ports:          &loadBalancer.Ports,
+			HealthCheck:    &loadBalancer.HealthCheck,
+		}
+
+		jobID, err := l.client.UpdateLoadBalancer(req)
+		if err != nil {
+			return err
+		}
+
+		onCompleteFn := func(loadBalancerID string) error {
+			loadBalancer, err := l.client.ReadLoadBalancer(loadBalancerID)
+			if err != nil {
+				return err
+			}
+
+			l.printer.StopSpinner()
+			return l.printer.PrintLoadBalancerHealthCheck(loadBalancer)
+		}
+
+		return l.waitOnJobHelper(c, jobID, "Updating health check on load balancer", onCompleteFn)
+
+	} else {
+		l.printer.Printf("No health check update required. Displaying load balancer health check.\n")
+		return l.printer.PrintLoadBalancerHealthCheck(loadBalancer)
+	}
+}
+
+func (l *LoadBalancerCommand) list(c *cli.Context) error {
+	loadBalancerSummaries, err := l.client.ListLoadBalancers()
+	if err != nil {
+		return err
+	}
+
+	return l.printer.PrintLoadBalancerSummaries(loadBalancerSummaries...)
+}
+
+func (l *LoadBalancerCommand) read(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "LOADBALANCER_NAME")
+	if err != nil {
+		return err
+	}
+
+	loadBalancer, err := l.client.ReadLoadBalancer(args["LOADBALANCER_NAME"])
+	if err != nil {
+		return err
+	}
+
+	return l.printer.PrintLoadBalancers(loadBalancer)
+}
+
+func parsePort(port, certificateName string) (*models.Port, error) {
+	split := strings.FieldsFunc(port, func(r rune) bool {
+		return r == ':' || r == '/'
+	})
+
+	if len(split) != 3 {
+		return nil, fmt.Errorf("Port format is: HOST_PORT:CONTAINER_PORT/PROTOCOL")
+	}
+
+	hostPort, err := strconv.ParseInt(split[0], 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("'%s' is not a valid integer", split[0])
+	}
+
+	containerPort, err := strconv.ParseInt(split[1], 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("'%s' is not a valid integer", split[1])
+	}
+
+	protocol := split[2]
+	if strings.ToLower(protocol) != "https" {
+		if certificateName != "" {
+			fmt.Printf("Warning: Certificate specified, but protocol of port is not HTTPS. Certificate will not be used.\n")
```

Comment:
I think this should actually be the inverse logic: we should check if the user specified `https` but the certificate is `""` and throw an error in that scenario. 

The reason I don't think we need this block is because you can only specify a certificate once, but can have multiple ports. You would always get warning if you did:
```
l0 loadbalancer create --certificate foo --port 443:443/https --port 80:80/http bar baz
``` 
even though that's a perfectly valid command.  

If you feel strongly we should keep this check, then it should be in a debug statement `log.Printf("[DEBUG] ...")` and specify which port is the culprit.

Reply:
I missed that case: that a user can specify multiple `--port` arguments in a single command. I'm going to just implement the error logic in that case rather than the warning logic.